lib/Sema/ConstraintSystem.h

```diff
@@ -4509,6 +4506,14 @@ class ConstraintSystem {
     return false;
   }
 
+  bool getExpressionTooComplex(SmallVectorImpl<Solution> const &solutions) {
+    size_t solutionMemory = 0;
+    for (auto const& s : solutions) {
```

Comment:
Nit: since like a good use-case for `std::accumulate`.

Reply:
I think using `std::accumulate` makes it harder to read:
```cpp
    size_t solutionMemory =
        std::accumulate(solutions.begin(), solutions.end(), 0,
            [](size_t memory, const Solution &solution) -> size_t {
              return memory + solution.getTotalMemory();
            });
    return getExpressionTooComplex(solutionMemory);
```
or
```cpp
    return getExpressionTooComplex(
        std::accumulate(solutions.begin(), solutions.end(), 0,
            [](size_t memory, const Solution &s) -> size_t {
              return memory + s.getTotalMemory();
            }));
```
What do you think?

Reply:
I'd prefer #2 but it really doesn't matter that much :)